src/protagonist/Engine/Ingest/Timebased/Transcode/ElasticTranscoder.cs

```diff
@@ -76,24 +78,29 @@ public class ElasticTranscoder : IMediaTranscoder
     {
         var asset = context.Asset;
         var assetId = context.AssetId;
-        var technicalDetails = asset.FullImageOptimisationPolicy.TechnicalDetails;
-        var outputs = new List<CreateJobOutput>(technicalDetails.Length);
+        var timeBasedPolicies = asset.ImageDeliveryChannels.Where(i => i.Channel == AssetDeliveryChannels.Timebased)
+            .Select(x => JsonSerializer.Deserialize<List<string>>(x.DeliveryChannelPolicy.PolicyData))
+            .SelectMany(i => i).ToList();

```

Comment:
In the future it might be worth encapsulating this query into a nicer extension method if the other policies use similar query to get values. I don't know if we know yet how the thumb + image policies will work to see how close they will be. Could allow for:

```cs
var timebasedPolicies = asset.ImageDeliveryChannels.GetTimebasedPolicyNames();
var thumbSizes = asset.ImageDeliveryChannels.GetThumbnailSizes();
```

I was wondering if we could map the `ImageDeliveryChannels` on way out of DB but that might make orchestrator/dapper access more difficult (e.g. `var timebasedPolicy = asset.ImageDeliveryChannels.OfType<TimebasedPolicy>();`) where `TimebasedPolicy` contains deserialized data.

Reply:
think this particular query is only used here, but I did add a `asset.GetThumbnailSizes()` method as that was used in a few places in engine/orchestrator - though looking at that, I will probably change it to sit on an `ImageDeliveryChannelsX` class, rather than `AssetX`